File: alura/Alura-Java/poo/bytebank-herdado/src/Gerente.java
public class Gerente extends Funcionario implements Autenticavel {

  private int senha;
  private final AutenticacaoUtil autenticador;

  public Gerente() {
    this.autenticador = new AutenticacaoUtil();
  }

  @Override
  public double getBonificacao() {
    System.out.println("Bonificando Gerente");
    return super.getSalario();
  }

  @Override
  public boolean autentica(int senha) {
    return this.autenticador.autentica(senha);
  }

  @Override
  public void setSenha(int senha) {
    this.autenticador.setSenha(senha);
  }
}

File: alura/Alura-Kotlin/kotlin/kotlin-orientacao-objetos/untitled/src/main/kotlin/Gerente.kt
class Gerente(
    nome:String,
    cpf: String,
    salario: Double,
    val senha: Int
) : Funcionario(nome, cpf, salario) {

    override val getBonificacao: Double get() = salario

    fun autentica(senha: Int): Boolean {
        if (this.senha == senha) {
            return true;
        }
        return false;
    }
}